source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/layout/FullscreenVideoLayout.java

```diff
@@ -265,149 +151,93 @@ public boolean onError(MediaPlayer mp, int what, int extra)
-        if (getCurrentState() == State.END)
-        {
-            stopCounter();
-        }
-    }
-
-    @Override
-    protected void tryToPrepare()
-    {
-        super.tryToPrepare();
-
-        if (getCurrentState() == State.PREPARED || getCurrentState() == State.STARTED)
-        {
-            if (m_elapsedTimeView != null && m_totalTimeView != null)
-            {
-                int total = getDuration();
-                if (total > 0)
-                {
-                    m_seekBar.setMax(total);
-                    m_seekBar.setProgress(0);
-
-                    total = total / 1000;
-                    long s = total % 60;
-                    long m = (total / 60) % 60;
-                    long h = (total / (60 * 60)) % 24;
-                    if (h > 0)
-                    {
-                        m_elapsedTimeView.setText("00:00:00");
-                        m_totalTimeView.setText(String.format(Locale.US, "%d:%02d:%02d", h, m, s));
-                    }
-                    else
-                    {
-                        m_elapsedTimeView.setText("00:00");
-                        m_totalTimeView.setText(String.format(Locale.US, "%02d:%02d", m, s));
-                    }
-                }
-            }
-
-            if (m_videoControlsView != null)
-            {
-                m_videoControlsView.setVisibility(View.VISIBLE);
-            }
-        }
+        m_mediaControlView.hide();
     }
 
     @Override
     public void start() throws IllegalStateException {
         if (!isPlaying())
         {
             super.start();
-            startCounter();
-            updateControls();
         }
+        updateControls();
     }
 
     @Override
     public void pause() throws IllegalStateException {
         if (isPlaying())
         {
-            stopCounter();
             super.pause();
-            updateControls();
         }
+        updateControls();
     }
 
     @Override
     public void reset()
     {
         super.reset();
-        stopCounter();
-        updateControls();
     }
 
     @Override
     public void stop() throws IllegalStateException
     {
         super.stop();
-        stopCounter();
         updateControls();
     }
 
     protected void updateControls()
     {
-        if (m_playButton == null) return;
-
-        Drawable icon;
-        if (getCurrentState() == State.STARTED)
+        if (m_currentState == State.STARTED)
         {
-            icon = m_context.getResources().getDrawable(android.R.drawable.ic_media_pause);
+            showControls();
         }
         else
         {
-            icon = m_context.getResources().getDrawable(android.R.drawable.ic_media_play);
+            showControlsForever();
         }
+    }
 
-        m_playButton.setBackground(icon);
+    public void hideControls()
+    {
+        m_mediaControlView.hide();
+        // return focus to player after hiding control view
+        requestFocus();
```

Comment:
is the hide and show methods still used or was the functionality of those functions replaced by this method (and the one below)?

Reply:
These three methods (`hideControls`, `showControls`, `showControlsForever`) are used instead of direct calls to show/hide.

The only exception is when releasing the player (in `releaseObjects` and `onDetachFromWindow`), then `.hide` is called directly.